### DISCUSSION TITLE
Set-ModuleReadMe script should not apply any updates (error)

### DISCUSSION THREAD

**Discussion by ThojoUno** [Azure/ResourceModules#3161] [General]
I get the error below when trying to run a modules pipeline. I have not made any updates to the module, or the modules readme.md file. Is there a step I'm missing when setting up the repo?

`##[error]    [-] [Microsoft.Resources/resourceGroups] Set-ModuleReadMe script should not apply any updates. 5.51s (5.51s|1ms)
##[error]     Expected $true, because The file hashes before and after applying the `Set-ModuleReadMe` function should be identical and should not have diff </br><pre>diff --git a/modules/Microsoft.Resources/resourceGroups/readme.md b/modules/Microsoft.Resources/resourceGroups/readme.md</br>index 56da440..d7efdee 100644</br>--- a/modules/Microsoft.Resources/resourceGroups/readme.md</br>+++ b/modules/Microsoft.Resources/resourceGroups/readme.md</br>@@ -259,6 +259,7 @@ module resourceGroups './Microsoft.Resources/resourceGroups/deploy.bicep' = {</br>     name: '<<namePrefix>>rrgmin001'</br>     // Non-required parameters</br>     enableDefaultTelemetry: '<enableDefaultTelemetry>'</br>+    tags: '<tags>'</br>   }</br> }</br> ```</br>@@ -282,6 +283,9 @@ module resourceGroups './Microsoft.Resources/resourceGroups/deploy.bicep' = {</br>     // Non-required parameters</br>     "enableDefaultTelemetry": {</br>       "value": "<enableDefaultTelemetry>"</br>+    },</br>+    "tags": {</br>+      "value": "<tags>"</br>     }</br>   }</br> }</pre>. Please re-run the script for this module's template., but got $false.
##[error]     at $filesAreTheSame | Should -Be $true -Because ('The file hashes before and after applying the `Set-ModuleReadMe` function should be identical and should not have diff </br><pre>{0}</pre>. Please re-run the script for this module''s template.' -f $mdFormattedDiff), /home/vsts/work/1/s/utilities/pipelines/staticValidation/module.tests.ps1:670
##[error]     at <ScriptBlock>, /home/vsts/work/1/s/utilities/pipelines/staticValidation/module.tests.ps1:670`

**Comment by u7x3611** [comment]
I also get same error.

**Comment by ThojoUno** [comment]
I had updated the test modules since our sandbox environment requires tags, so that must have triggered it. Solution is to dot source and run the Set-ModuleReadme.ps1 script against the main.bicep module.

. ./utilities/tools/Set-ModuleReadMe.ps1
Set-ModuleReadMe -TemplateFilePath 'C:\\Github\ResourceModules\modules\resources\resource-groups\main.bicep' -Verbose
